### PR TITLE
Implement basic network communication

### PR DESCRIPTION
**Pull request by elpozewaunig**
This PR implements basic network communication using SSE (Server-Sent Events). All functionality is unit-tested now.

### CODE REVIEW

**Review by Markus-M-AAU — APPROVED** [review]
**Excellent work!** 
*Only got some optional additions/questions that could possibly enhance the quality of the tests.*

In case they aren't trivially dismissed: For showcasing a working prototype in today's lecture we could skip these / branch them off for now.
`(Review for Unit-Test code only, as network-implementation was *partially* done in peer-programming and or worked through in detail by @elpozewaunig whilst writing the tests)`

File: src/test/java/com/se2gruppe5/risikobackend/sse/repositories/SseSinkRepositoryImplUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+                UUID.randomUUID(),
+                UUID.randomUUID()
+        );
+
+        for (int i = 0; i < uuids.size(); i++) {
+            sinkRepository.addSink(uuids.get(i), sinks.get(i));
+        }
+
+        // Check if getSinks() with all UUIDs returns all sinks
+        List<FluxSink<String>> returnedSinks = sinkRepository.getSinks(uuids);
```
[the remark this answers is not in the document]
Comment (elpozewaunig) [reply src/test/java/com/se2gruppe5/risikobackend/sse/repositories/SseSinkRepositoryImplUnitTest.java:101]: I don't quite understand what you mean here, we get sinks based on a list of uuids and in the next step we check if the returned list contains the same elements (since the order may differ from the original list). The sinkRepository doesn't store copies but the exact object (references).

[the remark this answers is not in the document]
Comment (elpozewaunig) [reply src/test/java/com/se2gruppe5/risikobackend/sse/repositories/SseSinkRepositoryImplUnitTest.java:101]: Ah, I see what you mean now. Sure, that can be done, but I'm also unsure whether we should actually require the list not being the same reference.

File: src/test/java/com/se2gruppe5/risikobackend/chat/controllers/ChatControllerUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import org.mockito.Mockito;
+
+class ChatControllerUnitTest {
+    private ChatController chatController;
+    private SseBroadcastService sseBroadcaster;
+
+    @BeforeEach
+    void setup() {
+        sseBroadcaster = Mockito.mock(SseBroadcastService.class);
+        chatController = new ChatController(sseBroadcaster);
```
[the remark this answers is not in the document]
Comment (elpozewaunig) [reply src/test/java/com/se2gruppe5/risikobackend/chat/controllers/ChatControllerUnitTest.java:15]: When a constructor is explicitly called, it can never return null, so this would be redundant ;).
https://stackoverflow.com/questions/11103444/can-constructor-return-a-null-object

File: src/test/java/com/se2gruppe5/risikobackend/sse/controllers/SseControllerUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import java.util.UUID;
+
+class SseControllerUnitTest {
+    private SseController sseController;
+    private SseBroadcastService sseBroadcaster;
+
+    @BeforeEach
+    void setup() {
+        sseBroadcaster = Mockito.mock(SseBroadcastService.class);
+        sseController = new SseController(sseBroadcaster);
```
[the remark this answers is not in the document]
Comment (elpozewaunig) [reply src/test/java/com/se2gruppe5/risikobackend/sse/controllers/SseControllerUnitTest.java:19]: Same as above ;)

File: src/test/java/com/se2gruppe5/risikobackend/sse/services/SseBroadcastServiceUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+
+class SseBroadcastServiceUnitTest {
+    private SseBroadcastService sseBroadcaster;
+    private SseSinkRepository mockSinkRepository;
+
+    @BeforeEach
+    void setup() {
+        mockSinkRepository = Mockito.mock(SseSinkRepository.class);
+        sseBroadcaster = new SseBroadcastService(mockSinkRepository);
```
[the remark this answers is not in the document]
Comment (elpozewaunig) [reply src/test/java/com/se2gruppe5/risikobackend/sse/services/SseBroadcastServiceUnitTest.java:22]: Same as above ;)

File: src/test/java/com/se2gruppe5/risikobackend/sse/services/SseBroadcastServiceUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+        Mockito.verify(mockSinkRepository).addSink(Mockito.any(UUID.class), Mockito.eq(sink));
+    }
+
+    @Test
+    void testRemoveSink() {
+        sseBroadcaster.removeSink(UUID.randomUUID());
+
+        Mockito.verify(mockSinkRepository).removeSink(Mockito.any(UUID.class));
+    }
+
```
[the remark this answers is not in the document]
Comment (elpozewaunig) [reply src/test/java/com/se2gruppe5/risikobackend/sse/services/SseBroadcastServiceUnitTest.java:66]: I'll add it, maybe it can prevent some edge cases of the same message being sent over and over :+1:

File: src/test/java/com/se2gruppe5/risikobackend/sse/repositories/SseSinkRepositoryImplUnitTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (elpozewaunig) [reply src/test/java/com/se2gruppe5/risikobackend/sse/repositories/SseSinkRepositoryImplUnitTest.java:103]: Done in the latest commit :)

File: src/test/java/com/se2gruppe5/risikobackend/sse/services/SseBroadcastServiceUnitTest.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (elpozewaunig) [reply src/test/java/com/se2gruppe5/risikobackend/sse/services/SseBroadcastServiceUnitTest.java:68]: Done in the latest commit :)

File: src/test/java/com/se2gruppe5/risikobackend/sse/repositories/SseSinkRepositoryImplUnitTest.java [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+        // Check if getSinks() with a few UUIDs returns the corresponding sinks
+        List<FluxSink<String>> returnedSubsetSinks = sinkRepository.getSinks(uuids.subList(0, 2));
+        List<FluxSink<String>> subsetSinks = sinks.subList(0, 2);
+        assertTrue(listsHaveSameElements(subsetSinks, returnedSubsetSinks));
+    }
+
+    // A simple equals would also check the order of the list elements
+    // However, the order does not matter in these tests
+    private <E> boolean listsHaveSameElements(List<E> list1, List<E> list2) {
```
[the remark this answers is not in the document]
Comment (elpozewaunig) [reply src/test/java/com/se2gruppe5/risikobackend/sse/repositories/SseSinkRepositoryImplUnitTest.java:118]: Done in the latest commit, as separate assertions :+1:
